unix/w0vncserver/portals/RemoteDesktop.cxx

```diff
@@ -499,3 +526,134 @@ bool RemoteDesktop::parseStreams(GVariant* streams)
 
   return true;
 }
+
+bool RemoteDesktop::loadRestoreToken()
+{
+  char filepath[PATH_MAX];
+  FILE* f;
+  const char* stateDir;
+  char restoreToken_[37];
+
+  // Allows users to reset the restore token by not choosing "Never"
+  if (askDisplayChoice != "Never")
+    clearRestoreToken();
+
+  if (askDisplayChoice == "Always")
+    return false;
+
+  // restoreToken will be empty the first time, we want to prompt the user
+  if (askDisplayChoice == "Once")
+    return !restoreToken.empty();
+
+  assert(askDisplayChoice == "Never");
+
+  // Only load from disk the first time
+  if (!restoreToken.empty())
+    return true;
+
+  stateDir = core::getvncstatedir();
+  if (!stateDir) {
+    vlog.error("Could not get state directory");
+    return false;
+  }
+
+  snprintf(filepath, sizeof(filepath), "%s/%s", stateDir, RESTORE_TOKEN_FILENAME);
+  f = fopen(filepath, "r");
+  if (!f) {
+    vlog.error("Could not open \"%s\": %s", filepath, strerror(errno));
+    return false;
+  }
+
+  if (fgets(restoreToken_, sizeof(restoreToken_), f)) {
+    uuid_t uuid;
+
+    fclose(f);
+    if (uuid_parse(restoreToken_, uuid) < 0) {
+      vlog.error("Invalid restore token, not a valid UUID string \"%s\"", restoreToken_);
+      return clearRestoreToken();
+    }
+    restoreToken = restoreToken_;
+    return true;
+  }
+  fclose(f);
+
+  vlog.error("Could not read restore token from \"%s\"", filepath);
+  return false;
+}
+
+bool RemoteDesktop::storeRestoreToken(const char* restoreToken_)
```

Comment:
You could also use a name like "newToken" to avoid weird underscore workarounds.